src/solver.py

```diff
@@ -0,0 +1,124 @@
+"""
+Collection of functions to solve for various inheritors.
+The function should take a tuple of inheritors, and return some shares.
+"""
+
+def solve(case: dict,
+          descendants:dict) -> dict:
+    """
+    This is the master solver. Individual inheritor solvers added to this
+    incrementally.
+    :param case: dictionary of inheritors and shares
+    :param descendants: dictionary mapping inheritor name to whether
+    or not they're a male/female descendant
+    :return: case with shares filled
+    """
+    case = solve_father(case=case,
+                        descendants=descendants)
+    case = solve_husband(case=case,
+                         descendants=descendants)
+    case = solve_wife(case=case,
+                      descendants=descendants)
+    case = solve_daughter(case=case)
+    case = solve_granddaughter(case=case)
+    #Add more here as we create more partial solvers
+    return case
+
+def solve_father(case: dict,
+                 descendants: dict)->dict:
+    """
+    Solve only for the father's share if he exists
+    :param case: dictionary of inheritors and shares
+    :param descendants: dictionary mapping inheritor name to whether
+    or not they're a male/female descendant
+    :return: string representing the fraction or asaba.
+    """
+    if 'father' not in case:
+        return case
+    #Check for any male descendants
+    if any([descendants[inh] == 'M' for inh in case]):
+        case['father']='1/6'
+    #Check for any female descendants
+    elif any([descendants[inh] == 'F' for inh in case]):
+        case['father'] = '1/6 + A'
+    else:
+        case['father'] = 'A'
+    return case
+
+def solve_husband(case: dict,
+                 descendants: dict)->dict:
+    """
+    Solve only for the father's share if he exists
+    :param case: dictionary of inheritors and shares
+    :param descendants: dictionary mapping inheritor name to whether
+    or not they're a male/female descendant
+    :return: string representing the fraction.
+    """
+    if 'husband' not in case:
+        return case
+    male_female_lookup={'M':0, 'F':0}
```

Comment:
```suggestion
    male_female_lookup=['M', 'F']
```
Sidi, would you be opposed to making this a list and a constant? List because the 0s in the dict threw me off for a second and constant because it's used twice

Reply:
No problem, used dict by habit only for the O(1) lookup but it makes no difference here, list will remain length 2.